### DISCUSSION TITLE
How to set the timeout of Kyuubi Sessions?

### DISCUSSION THREAD

**Discussion by HaoYang670** [apache/kyuubi#4725] [Q&A]
Hi! Is there a way to set the timeout for Kyuubi sessions? For example, if I run `spark.stop` to delete the Kyuubi engine, how long will be the Kyuubi sessions removed?

**Comment by yaooqinn** [comment]
yes, please check the doc

**Reply by HaoYang670** [reply]
Thank you. I've found `kyuubi.session.idle.timeout` to set the timeout of sessions.

But how could I let Kyuubi automatically delete sessions after the engine has been deleted?

**Reply by yaooqinn** [reply]
Isn't it automatically?

**Reply by HaoYang670** [reply]
No, if I use `spark.stop`, the sessions will be dangling.

**Reply by yaooqinn** [reply]
I mean timeout is not automatically?

**Reply by HaoYang670** [reply]
The timeout could work. But we don't want to set the timeout of **dangling** sessions same as the normal sessions.

What I want is that if a session can check whether the Kyuubi engine exists every once in a while. Is there a config such as `kyuubi.session.check.engine.exists.interval`?

**Reply by yaooqinn** [reply]
kyuubi.session.engine.alive.probe.enabled

**Reply by HaoYang670** [reply]
I have tried to set this property. But the session are not removed after I delete the cluster and wait for about 10 minutes
![image](https://user-images.githubusercontent.com/59198230/232992108-317a94f2-4777-49bb-ac2c-705bbef5a185.png)


Besides, I don't find a property to set the timeout of a session when the engine has been removed
![image](https://user-images.githubusercontent.com/59198230/232992473-e22e48c0-7375-410a-afa6-fbb87a24ded5.png)
Do I miss something?

**Comment by HaoYang670** [comment]
refer to #5060.
